### PR TITLE
Prevent filter on test overview to reveal old jobs

### PR DESCRIPTION
**Pull request by Martchus**
See https://progress.opensuse.org/issues/41756

### CODE REVIEW

**Comment by Martchus** [comment]
Unfortunately this required a bigger refactoring than I had expected. I had to move the filtering for results, states and failed modules out of the database query and implement it in Perl. This should not impair performance much since we're iterating over the data anyways. But hopefully this doesn't break any cases not covered by the tests.

File: lib/OpenQA/Utils.pm [hunk 1]
```diff
@@ -1247,5 +1246,29 @@ sub set_listen_address {
...
 
+sub param_hash {
+    my ($controller, $param_name) = @_;
+
+    my $params = $controller->every_param($param_name) or return;
+    my %hash;
+    for my $param (@$params) {
+        # ignore empty params
+        next unless $param;
+        # allow passing multiple values by seaparating them with comma
```
Comment (okurz) [inline lib/OpenQA/Utils.pm]: s/seaparating/separating/

Comment (coolo) [reply lib/OpenQA/Utils.pm]: I thought this was a reference to the old testament :)

**Comment by okurz** [comment]
+1

**Pull request merged**